### PR TITLE
Use @apollo/client

### PR DESCRIPTION
**Pull request by mllemango**
### WHY are these changes introduced?
Part of https://github.com/Shopify/first-party-library-planning/issues/15

Upgrading the retired apollo packages

🎩 'd, looks good on my end

### CODE REVIEW

File: client/_app.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-import ApolloClient from "apollo-boost";
-import { ApolloProvider } from "react-apollo";
-import { AppProvider as PolarisProvider } from "@shopify/polaris";
+import {
+  ApolloClient,
+  ApolloProvider,
+  InMemoryCache,
+  HttpLink,
+} from "@apollo/client";
+import { AppProvider as PolarisProvider, Page } from "@shopify/polaris";
```
Comment (paulomarg) [inline client/_app.js:8]: Do we need the `Page` import here? Doesn't seem like we're using it.

**Review by paulomarg — APPROVED** [review]
LGTM, haven't 🎩 ed it but it makes a lot of sense to me. We can probably test it out a bit further when we add an actual query to this.

File: client/_app.js (hunk 1 above)
Comment (mllemango) [reply client/_app.js:8]: ah 🤦‍♀️ oops didnt mean to leave that in there

**Comment by mllemango** [comment]
Just want to acknowledge that @hashiromer has a PR up accomplishing the same thing, but with the feature branch changes I've opened up a separate PR.

**Pull request merged**